peerprovider/http_test.go

```diff
@@ -0,0 +1,44 @@
+package peerprovider
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"golang.org/x/net/context"
+)
+
+func TestHTTPResolve(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		io.WriteString(w, "- 1.1.1.1:1\n")
+		io.WriteString(w, "- 2.2.2.2:2\n")
+	}))
+	defer svr.Close()
+
+	pp := httpPeerProvider{}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	peers, err := pp.Resolve(ctx, mustParseURL(svr.URL))
+	assert.NoError(t, err, "error resolving peers")
+	assert.Equal(t, peers, []string{
+		"1.1.1.1:1",
+		"2.2.2.2:2",
+	}, "received expected peers")
+}
+
+func TestHTTPResolveFails(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+	}))
+	defer svr.Close()
+
+	pp := httpPeerProvider{}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	_, err := pp.Resolve(ctx, mustParseURL(svr.URL))
+	assert.Error(t, err, "error resolving peers")
+}
```

Comment:
can we add another unit test where the `Get` fails? This can be done by picking a network address that will fail `e.g., 192.0.2.1`  which is for tests/examples?
http://smakd.potaroo.net/ietf/idref/rfc5735/index.html

maybe with a short timeout so the test doesn't take very long